sse/src/test/scala/akka/stream/alpakka/sse/EventSourceSpec.scala

```diff
@@ -0,0 +1,156 @@
+        Http(context.system).bindAndHandle(route, address, port).pipeTo(self)
+        context.become(binding)
+    }
+
+    private def binding: Receive = {
+      case serverBinding @ Http.ServerBinding(socketAddress) =>
+        log.info("Listening on {}", socketAddress)
+        context.system.scheduler.scheduleOnce(1500.milliseconds, self, Unbind)
+        context.become(bound(serverBinding))
+
+      case Status.Failure(cause) =>
+        log.error(cause, s"Can't bind to {}:{}!", address, port)
+        context.stop(self)
+    }
+
+    private def bound(serverBinding: Http.ServerBinding): Receive = {
+      case Unbind =>
+        serverBinding.unbind().map(_ => Done).pipeTo(self)
+        context.become(unbinding(serverBinding.localAddress))
+    }
+
+    private def unbinding(socketAddress: InetSocketAddress): Receive = {
+      case Done =>
+        log.info("Stopped listening on {}", socketAddress)
+        context.system.scheduler.scheduleOnce(500.milliseconds, self, Bind)
+        context.become(unbound)
+
+      case Status.Failure(cause) =>
+        log.error(cause, s"Can't unbind from {}!", socketAddress)
+        context.stop(self)
+    }
+  }
+
+  private def toServerSentEvent(setEventId: Boolean)(n: Int) = {
+    val id = n.toString
+    val event = ServerSentEvent(Some(id))
+    if (setEventId) event.copy(id = Some(id)) else event
+  }
+}
+
+class EventSourceSpec extends AsyncWordSpec with Matchers with BeforeAndAfterAll {
+  import EventSourceSpec._
+
+  private implicit val system = ActorSystem()
+  private implicit val ec = system.dispatcher
+  private implicit val mat = ActorMaterializer()
+
+  "EventSource" should {
+    "communicate correctly with an instable HTTP server" in {
+      val host = "localhost"
+      val port = 9999
+      val server =
+        actor(new Server(host, port, Server.route(setEventId = true)))
+      val nrOfSamples = 20
+      val events =
+        EventSource(Uri(s"http://$host:$port"), send, Some("2"))
+          .throttle(1, 500.milliseconds, 1, ThrottleMode.Shaping)
+          .take(nrOfSamples)
+          .runWith(Sink.seq)
+      val expected =
+        Seq.iterate(3, nrOfSamples)(_ + 1).map(toServerSentEvent(setEventId = true))
+      events.map(_ shouldBe expected).andThen({ case _ => system.stop(server) })
+    }
+
+    "apply the initial last event ID if the server doesn't set the event ID" in {
+      val host = "localhost"
+      val port = 9998
+      val server =
+        actor(new Server(host, port, Server.route(setEventId = false)))
+      val nrOfSamples = 20
+      val events =
+        EventSource(Uri(s"http://$host:$port"), send, Some("2")).take(nrOfSamples).runWith(Sink.seq)
+      val expected =
+        Seq.tabulate(20)(n => n % 2 + 3).map(toServerSentEvent(setEventId = false))
+      events.map(_ shouldBe expected).andThen({ case _ => system.stop(server) })
+    }
+  }
+
+  override protected def afterAll() = {
+    Await.ready(system.terminate(), 42.seconds)
```

Comment:
yeah, at least it doesn't make the CI build hang forever
(a realistic value is 10.seconds, but I'm still fine with 42)